probatus/feature_elimination/feature_elimination.py

```diff
@@ -695,3 +681,158 @@ def plot(self, show=True, **figure_kwargs):
+                keeping those columns.
+
+            min_features_to_select (int, optional):
+                Minimum number of features to be kept. This is a stopping criterion of the feature elimination. By
+                default the process stops when one feature is left. If columns_to_keep is specified in the fit method,
+                it may overide this parameter to the maximum between length of columns_to_keep the two.
+
+            cv (int, cross-validation generator or an iterable, optional):
+                Determines the cross-validation splitting strategy. Compatible with sklearn
+                [cv parameter](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.RFECV.html).
+                If None, then cv of 5 is used.
+
+            scoring (string or probatus.utils.Scorer, optional):
+                Metric for which the model performance is calculated. It can be either a metric name  aligned with predefined
+                [classification scorers names in sklearn](https://scikit-learn.org/stable/modules/model_evaluation.html).
+                Another option is using probatus.utils.Scorer to define a custom metric.
+
+            n_jobs (int, optional):
+                Number of cores to run in parallel while fitting across folds. None means 1 unless in a
+                `joblib.parallel_backend` context. -1 means using all processors.
+
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - nether prints nor warnings are shown
+                - 1 - 50 - only most important warnings
+                - 51 - 100 - shows other warnings and prints
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+
+            random_state (int, optional):
+                Random state set at each round of feature elimination. If it is None, the results will not be
+                reproducible and in random search at each iteration a different hyperparameters might be tested. For
+                reproducible results set it to integer.
+
+            early_stopping_rounds (int, optional):
+                Number of rounds with constant performance after which the model fitting stops. Only supported by some
+                models, such as xgboost and LightGBM.
+
+            eval_metric (str, optional):
+                Metric for early stopping.
+        """  # noqa
+        super(EarlyStoppingShapRFECV, self).__init__(clf, **kwargs)
+
+        if isinstance(self.clf, BaseSearchCV):
+            if self.verbose > 0:
+                warnings.warn(
+                    "EarlyStoppingShapRFECV is not compatible with hyperparameter tuning. "
+                    "The hyperparameter tuning step will be skipped. "
+                    "You can use ShapRFECV for that purpose."
+                )
+        self.search_clf = False
```

Comment:
Let's not set it to always false.

Maybe we can allow user to make hyperparam optimization before the CV SHap estimation,  but just warn him about the fact that it is not going to be used in calculating the performance and CV SHAP values.

The idea is:
You can optimize height of the tree without early stopping, etc,  but in the final thing it is going to run the optimized parameters already + early stopping.

And also overwriting it is not fully safe, because if someone passes RandomizedSearchCV, the clf is not going to extracted from it this way and there will be a very weird error.

So my proposal:
warning that: optimization is done only before without early stopping, and then in CV we use only early stopping. and not overwriting self.search_clf

What do you think?